src/libcore/result.rs

```diff
@@ -501,6 +501,8 @@ impl<T, E> Result<T, E> {
 
     /// Returns an iterator over the possibly contained value.
     ///
+    /// The iterator yields one value if the result is [`Ok`], otherwise none.
```

Comment:
Unless I'm misunderstanding it the iter will either have length 1 and contain the value inside the Ok result, or it will have length zero.  It never yields None, unless the result is Ok(None).  That's what I was trying to say by it yields 'no values at all'.

https://play.rust-lang.org/?gist=3d6a9500b782e878fae356a08888838b&version=stable&backtrace=0